datalad/support/gitrepo.py

```diff
@@ -2825,10 +2832,26 @@ def add_submodule(self, path, name=None, url=None, branch=None):
                 url = path
         cmd += [url, Path(path).as_posix()]
         self.call_git(cmd)
+
+        if subm is None:
+            # in case it did not happen above
+            subm = repo_from_path(self.pathobj / path)
```

Comment:
Why not just replace `subm = None` with this (and remove the assignment from under `if url is None`)?